tests/integration/e2e/RoomAdmin.test.tsx

```diff
@@ -369,4 +369,108 @@ describe('End-to-end Room Admin updating', function () {
+        });
+        context('when Room Name is not provided', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(roomNameInput,
+              { target: { value: '' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Room number is required to submit this form';
+            return waitForElement(
+              () => renderResult.getByText(errorMessage, { exact: false })
+            );
+          });
+        });
+        context('when Capacity is not provided', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(capacityInput,
+              { target: { value: '' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Capacity is required to submit this form, and it must be a positive number';
+            return waitForElement(
+              () => renderResult.getByText(errorMessage, { exact: false })
+            );
+          });
+        });
+        context('when Capacity is invalid', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(capacityInput,
+              { target: { value: '100z' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Capacity is required to submit this form, and it must be a positive number';
+            return waitForElement(
+              () => renderResult.getByText(errorMessage, { exact: false })
+            );
+          });
+        });
+      });
+      context('when the field values provided are valid', function () {
+        beforeEach(function () {
+          const modal = renderResult.getByRole('dialog');
+          roomNameInput = within(modal).getByLabelText('Room Number', { exact: false }) as HTMLInputElement;
+          capacityInput = within(modal).getByLabelText('Capacity', { exact: false }) as HTMLInputElement;
+          fireEvent.change(roomNameInput,
+            { target: { value: '513' } });
+          fireEvent.change(roomNameInput,
+            { target: { value: 100 } });
+        });
+        context('when the user tries to exit the modal', function () {
+          it('should show the unsaved changes warning', async function () {
+            const windowConfirmStub = stub(window, 'confirm');
+            windowConfirmStub.returns(true);
+            // Attempt to close the modal without saving
+            const cancelButton = await renderResult.findByText('Cancel');
+            fireEvent.click(cancelButton);
+            strictEqual(windowConfirmStub.callCount, 1);
+          });
+        });
+        context('when the form is submitted', function () {
```

Comment:
Would also be nice to have a test in here that confirms that updated Room appears in the list by searching for the update name/capacity.

Reply:
That's true - I ended up adding a test to make sure that the new room and capacity combination appears in a row of the table in commit [88a2f94](https://github.com/seas-computing/course-planner/pull/581/commits/88a2f949c959bee24be73d4e0e41652b9f28ece1).